src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs

```diff
@@ -43,20 +43,71 @@ class TrackedResourceListByImmediateParent : TypedRule<Dictionary<string, Schema
+            Regex pathRegEx = new Regex("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/[^/]+/[^/]+/{[^/]+}.*/(\\w+)$", RegexOptions.IgnoreCase);
             IEnumerable<KeyValuePair<string, Dictionary<string, Operation>>> filteredPaths = serviceDefinition.Paths.Where(path => pathRegEx.IsMatch(path.Key));
             return filteredPaths;
         }
 
+        private KeyValuePair<string, string> GetChildResourceName(string path, Dictionary<string, Dictionary<string, Operation>> paths, Dictionary<string, Schema> definitions)
+        {
+            Regex pathRegEx = new Regex("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/[^/]+/[^/]+/{[^/]+}.*/(\\w+)/{[^/]+}$", RegexOptions.IgnoreCase);
+            Match match = pathRegEx.Match(path);
+            KeyValuePair<string, string> result = new KeyValuePair<string, string>();
+            if (match.Success)
+            {
+                string childresourcename = match.Groups[1].Value;
+                string immediateparentresourcename_path = GetImmediateParentResourceName(path);
+                string immediateparentresourcename_actual = GetActualParentResourceName(immediateparentresourcename_path, paths, definitions);
+                result = new KeyValuePair<string, string>(childresourcename, immediateparentresourcename_actual);
+            }
+
+            return result;
+        }
+
+        private string GetActualParentResourceName(string nameinpath, Dictionary<string, Dictionary<string, Operation>> paths, Dictionary<string, Schema> definitions)
+        {
+            Regex pathRegEx = new Regex("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/.*/" + nameinpath + "/{[^/]+}$", RegexOptions.IgnoreCase);
+
+            IEnumerable<KeyValuePair<string, Dictionary<string, Operation>>> pths = paths.Where((KeyValuePair<string, Dictionary<string, Operation>> pth) => pathRegEx.IsMatch(pth.Key));
+            if (pths.Count() == 0) return null;
+            KeyValuePair<string, Dictionary<string, Operation>> path = pths.First();
+
+            IEnumerable<KeyValuePair<string, Operation>> operations = path.Value.Where(op => op.Key.Equals("get", StringComparison.CurrentCultureIgnoreCase));
+            if (operations.Count() == 0) return null;
+            KeyValuePair<string, Operation>  operation = operations.First();
+
+            IEnumerable<KeyValuePair<string, OperationResponse>> responses = operation.Value.Responses.Where(resp => resp.Key.Equals("200"));
+            if (responses.Count() == 0) return null;
+            KeyValuePair<string, OperationResponse> response = responses.First();
+
+            return GetReferencedModel(response.Value.Schema.Reference, definitions);
+        }
+
+        private string GetReferencedModel(String schema, Dictionary<string, Schema> definitions)
```

Comment:
Also, is there an example of this, do we need to walk the reference tree for what you're intending to use it?